<?xml version="1.0"?>                           <!--Declare XML Version typical value "1.0"-->

<!--Start a Robot description-->
<robot name="multiple_joints_example">         <!--Define a new robot name: Robot Name-->       

<!--Declare Links to be used-->
  <link name="link1" />
  <link name="link2" />
  <link name="link3" />
  <link name="link4" />
  <link name="link5" />
  <link name="link6" />

<!--Declare Joints to be used-->
    <joint name="joint_fixed" type="fixed"> <!--Declare Joint element 'name' and 'type' (revolute, continuos, prismatic, fixed)-->
        <parent link="link1"/>              <!--parent: Parent link name -->
        <child link="link2"/>               <!--child: child link name -->
        <origin xyz="1 1 1" rpy="0 0 0" />  <!--origin: This is the transform from the parent link to the child link xyz: xyz offset rpy:rotaton offset (radians)-->
    </joint>

    <joint name="joint_revolute" type="revolute"> <!--Declare Joint element 'name' and 'type' (revolute, continuos, prismatic, fixed)-->
        <parent link="link2"/>              <!--parent: Parent link name -->
        <child link="link3"/>               <!--child: child link name -->
        <origin xyz="0.5 0.5 0" rpy="0 0 0" />  <!--origin: This is the transform from the parent link to the child link xyz: xyz offset rpy:rotaton offset (radians)-->
        <axis xyz="0 0 1" />                <!--Rotation/Translation axis for the joints (revolute,continuos/prismatic) noy used with Fixed Joints -->
        <limit lower="-0.785" upper="0.785" effort="5.0" velocity="0.1"/> <!--Establish the limits of a joint Lower/Upper: lower/upper limits for revolute/prismatic joints in rad or m, Effort: maximum joint force (torque or force) depend on the joint, Velocity: Max velocity of the joint (rad/s or m/s) -->
    </joint>

    <joint name="joint_prismatic" type="prismatic">                 <!--Declare Joint element 'name' and 'type' (revolute, continuos, prismatic, fixed)-->
        <parent link="link1"/>                                      <!--parent: Parent link name -->
        <child link="link4"/>                                       <!--child: child link name -->
        <origin xyz="-1 -1 0" rpy="1.57 0 0" />                       <!--origin: This is the transform from the parent link to the child link xyz: xyz offset rpy:rotaton offset (radians)-->
        <axis xyz="1 0 0" />                                        <!--Rotation/Translation axis for the joints (revolute,continuos/prismatic) noy used with Fixed Joints -->
        <limit lower="-2" upper="2" effort="5.0" velocity="0.1"/>   <!--Establish the limits of a joint Lower/Upper: lower/upper limits for revolute/prismatic joints in rad or m, Effort: maximum joint force (torque or force) depend on the joint, Velocity: Max velocity of the joint (rad/s or m/s) -->
    </joint>

    <joint name="joint_continuous_1" type="continuous">
        <parent link="link4"/>
        <child link="link5"/>
        <origin xyz="0.5 0.5 0" rpy="0 0 0" />
        <axis xyz="1 0 0" />
    </joint>

    <joint name="joint_continuous_2" type="continuous">
        <parent link="link5"/>
        <child link="link6"/>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <axis xyz="0.707 0.707 0.707" />
    </joint>

</robot>
